contracts/rootchain/RootChain.sol

```diff
@@ -8,215 +8,287 @@ import "solidity-rlp/contracts/RLPReader.sol";
-        require(txList.length == 15, "incorrect tx list");
-        for(uint256 i = 0; i < 10; i++) {
-            require(txList[i].toUint() == 0, "incorrect tx fields");
-        }
-        require(txList[11].toUint() == msg.value, "mismatch in value");
-        require(txList[13].toUint() == 0, "second output must be zero'd");
-
-        /*
-            The signatures are kept seperate from the txBytes to avoid having to
-            recreate the txBytes for the confirmsig after both signatures are created.
-        */
+        depositNonce = depositNonce.add(1);
+    }
 
-        // construct the merkle root
-        bytes32 root = keccak256(txBytes);
-        uint256 position = getDepositBlock();
+    // @param depositNonce the nonce of the specific deposit
+    function startDepositExit(uint256 nonce)
+        public
+        payable
+    {
+        require(deposits[nonce].owner == msg.sender, "mismatch in owner");
+        require(depositExits[nonce].state == ExitState.NonExistent, "exit for this deposit already exists");
+        require(msg.value >= minExitBond, "insufficient exit bond");
+        if (msg.value > minExitBond) {
+            uint256 excess = msg.value - minExitBond;
+            balances[msg.sender] = balances[msg.sender].add(excess);
+            totalWithdrawBalance = totalWithdrawBalance.add(excess);
+        }
 
-        childChain[position] = childBlock({
-            root: root,
-            created_at: block.timestamp
+        uint amount = deposits[nonce].amount;
+        address owner = deposits[nonce].owner;
+        depositExitQueue.insert(nonce);
+        depositExits[nonce] = exit({
+            owner: owner,
+            amount: amount,
+            created_at: block.timestamp,
+            state: ExitState.Pending
         });
 
-        currentDepositBlock = currentDepositBlock.add(1);
-        emit Deposit(txList[10].toAddress(), msg.value);
+        emit StartedDepositExit(nonce, owner, amount);
     }
 
-
-    /// @param txPos [0] Plasma block number in which the transaction occured
-    /// @param txPos [1] Transaction Index within the block
-    /// @param txPos [2] Output Index within the transaction (either 0 or 1)
-    /// @param sigs First 130 bytes are signature of transaction and the rest is confirm signature
-    /// @notice Each signature is 65 bytes
-    function startExit(uint256[3] txPos, bytes txBytes, bytes proof, bytes sigs)
+    // Transaction encoding:
+    // [Blknum0, TxIndex0, Oindex0, depositNonce0, Amount0, ConfirmSig0
+    //  Blknum1, TxIndex1, Oindex1, depositNonce1, Amount1, ConfirmSig1
+    //  NewOwner0, Denom0, NewOwner1, Denom1, Fee]
+    //
+    // @param txPos   location of the transaction [blkNum, txIndex, outputIndex]
+    // @param txBytes raw transaction bytes
+    // @param proof   merkle proof of inclusion in the child chain
+    // @param sigs    signatures of transaction including confirm signatures
```

Comment:
sigs no longer include confirmSigs correct? If so, this should be changed to avoid confusion

Reply:
ah yes